libpod/networking_linux.go

```diff
@@ -225,21 +226,55 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) error {
 
 	isSlirpHostForward := false
 	disableHostLoopback := true
+	var outboundAddr string
+	outboundAddr = nil
+
+	var outboundAddr6 string
+	outboundAddr6 = nil
+
 	if ctr.config.NetworkOptions != nil {
 		slirpOptions := ctr.config.NetworkOptions["slirp4netns"]
 		for _, o := range slirpOptions {
 			switch o {
 			case "port_handler=slirp4netns":
 				isSlirpHostForward = true
+				continue
 			case "port_handler=rootlesskit":
 				isSlirpHostForward = false
+				continue
 			case "allow_host_loopback=true":
 				disableHostLoopback = false
+				continue
 			case "allow_host_loopback=false":
 				disableHostLoopback = true
+				continue
+			}
+			// Parametrized options
+			switch {
+			case slirpFeatures.HasOutboundAddr && strings.Contains(o, "outbound_addr="):
+				ipv4OrInterface := string(o[14:])
+				ipv4 := net.ParseIP(ipv4OrInterface)
+				if ipv4 == nil || ipv4.to4() == nil {
+					intercace, err := net.InterfaceByName(ipv4OrInterface)
+					if err != nil {
+						return errors.Wrapf(err, "invalid outbound_addr")
+					}
+				}
+				outboundAddr = ipv4OrInterface
+				continue
+			case slirpFeatures.HasOutboundAddr && strings.Contains(o, "outbound_addr6="):
+				ipv6OrInterface := string(o[15:])
+				ipv6 := net.ParseIP(ipv6OrInterface)
+				if ipv6 == nil || ipv6.to4() != nil {
```

Comment:
The default string type of golang is null, which is "", can not use nil in string

Reply:
fixed